at_secondary/at_secondary_server/lib/src/exception/global_exception_handler.dart

```diff
@@ -152,6 +159,10 @@ class GlobalExceptionHandler {
 
   void _writeToSocket(AtConnection atConnection, String prompt,
       String? errorCode, String errorDescription) {
-    atConnection.write('error:$errorCode-$errorDescription\n$prompt');
+    var errorJsonMap = {
```

Comment:
Won't an old client fail to parse the JSON?

Reply:
This is taken care in at_lookup library.

In at_lookup, On receiving the error response, tries to JSON decode the error response. If FormatException is thrown, exception is caught and error response is parsed in the older way.

Reply:
Yes it's taken care of ... IF you have the new at_lookup library

My point is that old clients will break. This is a fundamentally breaking change

I believe we need a way for a client to let a server know some things about itself - eg what version of client software am I running, and maybe a set of "features the client supports"

Reply:
Let's discuss on architecture call